src/events/websocket/HttpServer.js

```diff
@@ -24,6 +26,14 @@ export default class HttpServer {
       },
     }
 
+    // HTTPS support
+    if (typeof httpsProtocol === 'string' && httpsProtocol.length > 0) {
+      serverOptions.tls = {
+        cert: readFileSync(resolve(httpsProtocol, 'cert.pem'), 'ascii'),
```

Comment:
I refactored applying the TLS options with the non-blocking `node:fs/promise` versions if you could base it on that if possible?